### DISCUSSION TITLE
Derive status of a device which is "gone"

### DISCUSSION THREAD

**Discussion by FrankH1** [ESPresense/ESPresense#1269] [Ideas]
Hi,
I'm doing my fist steps with ESPresense. Have installed 3.3.2 on a 8685WROOM-06-H4. Works stable (once I figured out that the power from my USB-Programming interface is not enough and I have to supply the module from a more powerful source). Have set-up MQTT and doing some tests. Finally the messages will go into my OpenHab-smarthome.
I'm looking for an easy way to figure out, if a device is present at home or not.
My problem - once the device is in reach  I see the MQTT messages with the distance, rssi etc. So I can easily figure out, when the owner of the device enters the house. But I don't see when the device is leaving- the only hint I get is that I just don't get new MQTT messages. But there must be already something in ESPresense - because the webterminal (https://espresense.com/terminal) shows the ID with "Del" after the device is gone (probably based on some timers or something more sophisticated). Is it possible to get this info via MQTT? (tbh - I don't want to program monitoring rules which check the status every x seconds - and if there is no new MQTT - than obviously the device is gone - if this is already done by the ESP)
Have someone got an idea?
Thanks Frank

**Comment by FrankH1** [comment]
To be more precise: 
I'm looking for the information like it is delivered by the web-terminal regarding devices, so the Information in the lines starting with "New...." and "Del ...." - and want to get that Information by MQTT. I'm only interested in a yes/no answer for presence of monitored devices/beacons.
E.g. one MQTT message "online" if a monitored beacon is discovered (= new in the web-terminal) and one MQTT message "offline" if a monitored beacon is out of range/ gone (=del in the web-terminal). The information is obviously there (because it's visible in web-terminal) but I don't find it in MQTT messages.
Best regards

**Comment by rackilucky** [comment]
Same here :)

**Comment by FrankH1** [comment]
Meanwhile sorted. Have the items which get the distance via MQTT and did set an expiration timer for the item (10 sec) in OH. This sets the distance to an arbirtrary value of 100 m if no update happens. The distance is than checked by a simple rule

rule "Bluetooth Frank"
when Item ESPresense_PhoneFrankDistance changed
then
	PhoneOnlineFrankBlue.postUpdate(ON)
	if (ESPresense_PhoneFrankDistance.state  > 99) PhoneOnlineFrankBlue.postUpdate(OFF)
end

So an additional item holds the online state. Was at the end quite simple, because OH supports expiration timers without doing a lot of coding.

**Reply by rackilucky** [reply]
is it guaranteed that every devices sends messages within 10sec as long as it is in range ?
I thought the delay between update messages can be up to minutes (if device in sleep mode or so)

**Reply by FrankH1** [reply]
Don't know. I'm using a Beacon with an advertising rate of 1 HZ. So every second one update - and I found that even if the device is unmoved and laying somewhere the distance changes by a small amount.
